jbcli/jbcli/utils/dockerutil.py

```diff
@@ -233,33 +224,87 @@ def pull(tag):
-            pushed = pushed + datetime.timedelta(hours=5) # Why do we add 5 hours?
             for tag in image['imageTags']:
-                slang = '{} days ago'.format((now - pushed).days)
-                if not showall and not semantic:
-                    if pushed >= now - datetime.timedelta(days=30):
-                        imageList.append(
-                            [tag, image['imageDigest'][7:], slang])
-                elif showall and not semantic:
-                    imageList.append(
-                        [tag, image['imageDigest'][7:], slang])
-                elif semantic and not showall:
-                    if '.' in tag:
-                        imageList.append([tag, image['imageDigest'][7:],
-                                          slang])
+                human_readable = human_readable_timediff(pushed)
+                is_semantic_tag = bool(semantic_version_tag_pattern.match(tag))
+                if tag == 'master':
+                    tag_priority = 4
+                elif tag == 'develop': 
+                    tag_priority = 3
+                elif is_semantic_tag:
+                    tag_priority = 2
+                else:
+                    tag_priority = 1
+
+                # If semantic, just return semantic tags
+                meets_semantic_criteria = (semantic and is_semantic_tag) or not semantic
+                # Use if showall is true, return all tags, otherwise return just last 30 days
+                meets_time_criteria = showall or (pushed >= now - datetime.timedelta(days=30))
+
+                if meets_semantic_criteria and meets_time_criteria:
+                    row = [tag, pushed, human_readable, tag_priority, is_semantic_tag]
+                    imageList.append(row)
+
+    # Find the semantic version of stable
+    imageList.sort(key=itemgetter(1))
+
+    newImageList = []
+    prevtag = None
+    for row in imageList:
+        tag, pushed, human_readable, tag_priority, is_semantic_tag = row
+        if tag == 'master':
+            row = [tag, pushed, human_readable, tag_priority, is_semantic_tag, prevtag]
+        else:
+            row = [tag, pushed, human_readable, tag_priority, is_semantic_tag, None]
+        newImageList.append(row)
+        prevtag = tag
+
+    # Sort in descending order of priority and timestamp
+    def sort_order(row):
+        return (row[3], row[0])
+    newImageList.sort(key=sort_order, reverse=True)
 
     if print_flag:
-        print(tabulate(sorted(imageList, key=itemgetter(0)), headers=['Image Name', 'Digest',
-                                           'Time Tagged']))
+        print(
+            tabulate(newImageList,
+                headers=['Image Name', 'Time Tagged', 'Relative', 'Priority', 'Is Semantic', 'Master Version']
+            )
+        )
+
+    return newImageList
+
+
+def set_tag(env, tag):
+    """Set an environment to use a tagged image """
+    ensure_root()
```

Comment:
We should get rid of `ensure_root`. Instead, figure out the `devlandia` directory based on the python module (`__file__`) and then find the docker-compose.yml relative to that.